src/core-layers/hexagon-layer/hexagon-layer.js

```diff
@@ -347,13 +347,11 @@ export default class HexagonLayer extends CompositeLayer {
   // for subclassing, override this method to return
   // customized sub layer props
   getSubLayerProps() {
-    const {id, radius, elevationScale, extruded, coverage, lightSettings, fp64} = this.props;
-
-    const forwardProps = this.getBaseLayerProps();
+    const {radius, elevationScale, extruded, coverage, lightSettings, fp64} = this.props;
 
     // return props to the sublayer constructor
-    return Object.assign({}, forwardProps, {
-      id: `${id}-hexagon-cell`,
+    return super.getSubLayerProps({
+      id: 'hexagon-cell',
```

Comment:
I think id is necessary, when you have multiple Hexagon-layer, each hexagon-cell layer should have its unique id

Reply:
getSubLayerProps() does this logic automatically.

Reply:
> I think id is necessary, when you have multiple Hexagon-layer, each hexagon-cell layer should have its unique id

This new method automatically prepends the parent layer id I think. That needs to be clear in the docs.

Reply:
i see, make sense